src/views/mainView.jsx

```diff
@@ -26,7 +28,11 @@ export const ConferenceMainView = ({
   const peerSharingPlaylist = useHMSStore(selectPeerSharingVideoPlaylist);
   const roomState = useHMSStore(selectRoomState);
   const hmsActions = useHMSActions();
-  const { audioPlaylist, videoPlaylist, uiViewMode } = useContext(AppContext);
+  const { audioPlaylist, videoPlaylist, uiViewMode, roomMetadata } =
+    useContext(AppContext);
+  let HLSViewerRole =
+    roomMetadata[DEFAULT_HLS_ROLE_KEY] ||
```

Comment:
Instead of having this at every place let the context return hlsViewerRole which can be directly used